pkgs/build-support/agda/default.nix

```diff
@@ -70,8 +72,8 @@ let
         installPhase = if installPhase != null then installPhase else ''
           runHook preInstall
           mkdir -p $out
-          find \( ${concatMapStringsSep " -or " (p: "-name '*.${p}'") (extensions ++ extraExtensions)} \) -exec cp -p --parents -t "$out" {} +
-          runHook postInstall
+          find -not \( -path ${everythingFile} -or -path ${interfaceFile everythingFile} \) -and \( ${concatMapStringsSep " -or " (p: "-name '*.${p}'") (extensions ++ extraExtensions)} \) -exec cp -p --parents -t "$out" {} +
```

Comment:
Is this necessary? If the `Everything.agda` file is actually called `Everything.agda.tex` then the `everythingFile` variable should be `Everything.agda.tex` right?

Reply:
IIUC yes

Reply:
Is what necessary?

Reply:
the `-path ${interfaceFile everythingFile}` bit. In any case this doesn't look like an interface file?

Reply:
The interface function should return the path to the interface file, e.g.

```
interfaceFile "src/Everything.lagda.tex" == "src/Everything.agdai"
```

The intention is that we don't copy the `everythingFile` nor the compiled interface file.

Reply:
ah right that makes sense

Reply:
Is there a way to write some sort of unit tests in nixpkgs?

Reply:
I am not sure unfortunately. Not that I know of though

Reply:
Maybe I should comment `interfaceFile` to clarify the intention.

Reply:
You could use the option in `nixos/tests/agda.nix` though :shrug:

Reply:
I'll do that.